provider/manual/provider.go

```diff
@@ -57,6 +60,35 @@ func (p manualProvider) CloudSchema() *jsonschema.Schema {
 	return cloudSchema
 }
 
+// Ping tests the connection to the cloud, to verify the endpoint is valid.
+func (p manualProvider) Ping(endpoint string) error {
+	// do a normal TCP dial first, because it makes the error messages clearer
+	// when things fail.
+	addr := net.JoinHostPort(endpoint, "22")
+	c, err := net.Dial("tcp", addr)
+	if err != nil {
+		return errors.Errorf("Can't connect to %s: "+err.Error(), addr)
+	}
+	c.Close()
```

Comment:
Nah, this is just testing that we can connect. If we can, we don't need the connection, so we just close it immediately.